pybop/parameters/priors.py

```diff
@@ -90,6 +122,98 @@ def rvs(self, size=1, random_state=None):
             loc=self.loc, scale=self.scale, size=size, random_state=random_state
         )
 
+    def __call__(self, x):
```

Comment:
I would suggest removing the `call`, `evaluate` and `_evaluate` functions from BasePrior because different values are needed across the code (e.g. `pdf`, `logpdf`), it is better to be specific about what is being evaluated here. The evaluateS1 function could be called `logpdfS1`